Doc/library/asyncio-stream.rst

```diff
@@ -206,12 +206,21 @@ StreamReader
 
    .. coroutinemethod:: read(n=-1)
 
-      Read up to *n* bytes.  If *n* is not provided, or set to ``-1``,
-      read until EOF and return all read bytes.
+      Read up to *n* bytes from the stream.
 
+      If *n* is not provided, or set to ``-1``,
+      read until EOF, then return all read :class:`bytes`.
       If EOF was received and the internal buffer is empty,
       return an empty ``bytes`` object.
 
+      If *n* is ``0``, return an empty ``bytes`` object immediately.
+
+      If *n* is positive, wait for at least 1 byte to become available
+      (or EOF), then return the available ``bytes`` which might be less
+      than *n*.
```

Comment:
Unlike the previous, this appears to neglect to explicitly mention anything about what the value of `n` actually does here (the number of bytes to return, at most). Adding that back, eliding the now unnecessary mention that fewer bytes may be returned (since that's what "at most" means), implementing Guido's suggestion and fixing a couple other things, we have the much simplified

```suggestion
      If *n* is positive, return at most *n* available ``bytes``
      as soon as at least 1 byte is available in the internal buffer.
```